### PR TITLE
add cognito (jwt bearer) authentication

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary
Add Cognito token verification

## ⚙️ Test Data and/or Report
See test procedure on the wiki 
https://github.com/NASA-PDS/doi-service/wiki/Test-API-with-Cognito-authentication

## ♻️ Related Issues

fixes https://github.com/NASA-PDS/doi-service/issues/348

### CODE REVIEW

**Comment by tloubrieu-jpl** [comment]
Hi @ramesh-maddegoda , I was able to make the decode function also verify the issuer. That should be ok to start with that update. Thanks again for your help on that.

File: src/pds_doi_service/api/controllers/authentication.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from jose import JWTError, jwt
+from flask import current_app
+from werkzeug.exceptions import Unauthorized
+from pds_doi_service.core.util.config_parser import DOIConfigUtil
+
+config = DOIConfigUtil().get_config()
+
+JWT_ISSUER = config.get('API_AUTHENTICATION', 'jwt_issuer')
+JWT_SECRET = config.get('API_AUTHENTICATION', 'jwt_secret')
+JWT_LIFETIME_SECONDS = 600
```
Comment (ramesh-maddegoda) [inline src/pds_doi_service/api/controllers/authentication.py:12]: @tloubrieu-jpl , is the JWT_LIFETIME_SECONDS variable used? If this is a life time of a JWT access token, then the default value should be 3600 seconds.

File: src/pds_doi_service/api/controllers/authentication.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+def decode_token(token):
+    try:
+        current_app.logger.debug("try to decode/validate token %s", token)
+        return jwt.decode(
+            token,
+            JWT_SECRET,
+            algorithms=[JWT_ALGORITHM],
+            issuer=JWT_ISSUER,
+            options={
+                'verify_signature': False,
```
Comment (ramesh-maddegoda) [inline src/pds_doi_service/api/controllers/authentication.py:25]: Can we make it work with 'verify_signature': True?

File: src/pds_doi_service/api/swagger/swagger.yaml [hunk 3]
```diff
@@ -125,6 +125,8 @@ paths:
         for one or more  records to be submitted. Record URLs must resolve to PDS4
         label files (xml).
       operationId: post_dois
+      security:
+      - jwt: ['']
```
Comment (nutjob4life) [inline src/pds_doi_service/api/swagger/swagger.yaml:129]: When I see a colon on a line above, I reflexively want to see an indentation on the next line:
```yaml
security:
  - jwt: ['']
```
Although `yamllint` says they evaluate to the same data structure. (Perhaps I program too much Python? 😁)

**Review by nutjob4life — APPROVED** [review]

**Review by nutjob4life — APPROVED** [review]
👍

**Comment by jordanpadams** [comment]
@ramesh-maddegoda do you mind taking over this PR and making the necessary changes for @tloubrieu-jpl?

File: src/pds_doi_service/api/swagger/swagger.yaml (hunk 3 above)
Comment (ramesh-maddegoda) [reply src/pds_doi_service/api/swagger/swagger.yaml:129]: @nutjob4life, the Swagger formatting in the above line 128 and 129 is consistent with the format of rest of the content in the file and also with YAML files generated by AWS API Gateway etc. So, I think it is ok to keep it as it is.

File: src/pds_doi_service/api/controllers/authentication.py (hunk 1 above)
Comment (ramesh-maddegoda) [reply src/pds_doi_service/api/controllers/authentication.py]: I updated it

File: src/pds_doi_service/api/controllers/authentication.py (hunk 2 above)
Comment (ramesh-maddegoda) [reply src/pds_doi_service/api/controllers/authentication.py]: Updated code to verify_signature

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**